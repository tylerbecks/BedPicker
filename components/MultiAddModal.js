import React, { useState, useEffect, useRef } from "react";
import Modal from "react-native-modal";
import {
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View
} from "react-native";

const styles = StyleSheet.create({
  addButton: {
    alignItems: "flex-end"
  },
  addButtonText: {
    color: "#14aaf5",
    fontSize: 16
  },
  addButtonTextDisabled: {
    color: "#b7bfc6",
    fontSize: 16
  },
  modal: {
    justifyContent: "flex-end",
    margin: 0
  },
  modalContent: {
    backgroundColor: "white",
    borderRadius: 10,
    flexDirection: "row",
    alignItems: "flex-start",
    justifyContent: "space-between",
    height: "50%",
    padding: 20
  },
  textInput: {
    marginBottom: 10
  }
});

export default ({ onSubmit, isVisible, close, placeholder }) => {
  const [text, setText] = useState("");
  const textInput = useRef(null);

  const handleInputBlur = () => {
    if (!textInput.current) {
      return;
    }
    if (!isVisible) {
      textInput.current.blur();
      return;
    }

    setTimeout(() => {
      textInput.current.focus();
    }, 20);
  };

  useEffect(() => {
    handleInputBlur();
  });

  const handleSubmitEditing = () => {
    onSubmit(text);
    setText("");
  };

  const handleChangeText = newText => {
    setText(newText);
  };

  const isAddButtonDisabled = () => text.length === 0;

  return (
    <Modal
      isVisible={isVisible}
      onBackdropPress={close}
      onSwipe={close}
      swipeDirection="down"
      style={styles.modal}
      backdropOpacity={0.5}
    >
      <View style={styles.modalContent}>
        <TextInput
          ref={textInput}
          autoCapitalize="words"
          blurOnSubmit={false}
          enablesReturnKeyAutomatically
          onChangeText={handleChangeText}
          onSubmitEditing={handleSubmitEditing}
          placeholder={placeholder}
          value={text}
          style={styles.textInput}
        />
        <TouchableOpacity
          onPress={handleSubmitEditing}
          style={styles.addButton}
          disabled={isAddButtonDisabled()}
        >
          <Text
            style={
              isAddButtonDisabled()
                ? styles.addButtonTextDisabled
                : styles.addButtonText
            }
          >
            Add
          </Text>
        </TouchableOpacity>
      </View>
    </Modal>
  );
};
