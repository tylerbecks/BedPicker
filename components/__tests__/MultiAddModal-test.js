import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MultiAddModal from '../MultiAddModal';

it('renders', () => {
  expect(
    renderer
      .create(
        <MultiAddModal
          onSubmit={jest.fn()}
          close={jest.fn()}
          placeholder="foo"
          isVisible
        />
      )
      .toJSON()
  ).toMatchSnapshot();
});
