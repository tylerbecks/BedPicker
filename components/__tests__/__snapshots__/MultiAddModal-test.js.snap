// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onRequestClose={[Function]}
  onSwipe={[MockFunction]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection="down"
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onSwipe={[MockFunction]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
        "left": 0,
        "margin": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection="down"
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "height": "50%",
          "padding": 20,
        }
      }
    >
      <TextInput
        autoCapitalize="words"
        blurOnSubmit={false}
        enablesReturnKeyAutomatically={true}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="foo"
        style={
          Object {
            "marginBottom": 10,
          }
        }
        value=""
      />
    </View>
  </View>
</Modal>
`;
